zipline/pipeline/loaders/blaze/_core.pyx

```diff
@@ -524,6 +524,13 @@ cdef array_for_column(object dtype,
         raise TypeError('unknown column dtype: %r' % input_array.dtype)
 
 
+cpdef str getname(object column):
+    try:
+        return column.metadata['blaze_column_name']
```

Comment:
column.metadata always exists, it was added in: https://github.com/quantopian/zipline/commit/cd09c5bc5c2a2f626855fec307ebd80d8fe532d8#diff-ca55a97f3934c34c38c2be0840a85340R178`